doc/man3/SSL_CTX_set_stateless_cookie_generate_cb.pod

```diff
@@ -20,28 +22,48 @@ SSL_CTX_set_stateless_cookie_verify_cb
      int (*verify_stateless_cookie_cb) (SSL *ssl,
                                         const unsigned char *cookie,
                                         size_t cookie_len));
+ void SSL_CTX_set_cookie_generate_cb(
+     SSL_CTX *ctx,
+     int (*app_gen_cookie_cb)(SSL *ssl,
+                              unsigned char *cookie,
+                              unsigned int *cookie_len));
+ void SSL_CTX_set_cookie_verify_cb(
+     SSL_CTX *ctx,
+     int (*app_verify_cookie_cb)(SSL *ssl,
+                                 const unsigned char *cookie,
+                                 unsigned int cookie_len));
 
 =head1 DESCRIPTION
 
-SSL_CTX_set_cookie_generate_cb() sets the callback used by L<SSL_stateless(3)>
+SSL_CTX_set_stateless_cookie_generate_cb() sets the callback used by L<SSL_stateless(3)>
 to generate the application-controlled portion of the cookie provided to clients
 in the HelloRetryRequest transmitted as a response to a ClientHello with a
 missing or invalid cookie. gen_stateless_cookie_cb() must write at most
 SSL_COOKIE_LENGTH bytes into B<cookie>, and must write the number of bytes
 written to B<cookie_len>. If a cookie cannot be generated, a zero return value
 can be used to abort the handshake.
 
-SSL_CTX_set_cookie_verify_cb() sets the callback used by L<SSL_stateless(3)> to
+SSL_CTX_set_stateless_cookie_verify_cb() sets the callback used by L<SSL_stateless(3)> to
 determine whether the application-controlled portion of a ClientHello cookie is
-valid. A nonzero return value from app_verify_cookie_cb() communicates that the
+valid. A nonzero return value from verify_stateless_cookie_cb() communicates that the
 cookie is valid. The integrity of the entire cookie, including the
 application-controlled portion, is automatically verified by HMAC before
 verify_stateless_cookie_cb() is called.
 
+If you want to do cookies in DTLS, you have to use the older functions SSL_CTX_set_cookie_generate_cb()
+and SSL_CTX_set_cookie_verify_cb(). These are only for use in DTLS applications (not TLS1.2).
+See the DTLSv1_listen() documents for some discussion of that.
```

Comment:
Probably we need to at least describe the arguments to these functions. It's actually very similar to the "stateless" version except the max cookie length is `DTLS1_COOKIE_LENGTH` instead of `SSL_COOKIE_LENGTH` and there is no integrity checking.

So perhaps something like:
````
SSL_CTX_set_cookie_generate_cb() sets the callback used in DTLS by the
L<DTLSv1_listen(3)> function
to generate the application-controlled portion of the cookie provided to clients
in the HelloVerifyRequest transmitted as a response to a ClientHello with a
missing or invalid cookie. app_gen_cookie_cb() must write at most
DTLS1_COOKIE_LENGTH bytes into B<cookie>, and must write the number of bytes
written to B<cookie_len>. If a cookie cannot be generated, a zero return value
can be used to abort the handshake.

SSL_CTX_set_cookie_verify_cb() sets the callback used by L<DTLSv1_listen(3)> to
determine whether the cookie in the ClientHello is
valid. A nonzero return value from app_verify_cookie_cb() communicates that the
cookie is valid. No integrity checking of the cookie is verified before this function
is called.
````